src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala

```diff
@@ -136,6 +159,53 @@ class AppsController(
 
   private def patchSingle(appId: PathId)(implicit identity: Identity): Route = ???
 
+  private[this] def updateMultiple(partialUpdate: Boolean, allowCreation: Boolean)(implicit identity: Identity): Route = {
+    val version = clock.now()
+    (forceParameter & entity(as(appUpdatesUnmarshaller(partialUpdate)))) { (force, appUpdates) =>
+      def updateGroup(rootGroup: RootGroup): RootGroup = appUpdates.foldLeft(rootGroup) { (group, update) =>
```

Comment:
How about moving this method out and test it separately? I actually thought that we mix a lost of controller and service/model logic. Ideally the controllers call one method or so. I haven't been following myself, though.